app/src/main/java/io/neurolab/activities/PinLayoutActivity.java

```diff
@@ -33,16 +46,56 @@ protected void onCreate(Bundle savedInstanceState) {
 
         boolean frontSide = getIntent().getBooleanExtra("layout", true);
         ImageView pinImageView = findViewById(R.id.pin_lay_img_view);
-        pinImageView.setImageDrawable(frontSide ? getDrawable(R.drawable.front_pin_layout) : getDrawable(R.drawable.back_pin_layout));
+        colorMap = findViewById(R.id.img_neurolab_color_map);
+        colorMap.setImageDrawable(ResourcesCompat.getDrawable(getResources(),
+                frontSide ? R.drawable.neurolab_front_colormap : R.drawable.neurolab_back_colormap, null));
+        pinImageView.setImageDrawable(ResourcesCompat.getDrawable(getResources(), frontSide ? R.drawable.front_pin_layout : R.drawable.back_pin_layout, null));
 
         pinImageView.setOnTouchListener(this);
+
+        populatePinDetails();
+    }
+
+    private void populatePinDetails() {
+        pinDetails.add(new PinDetails("ESP", "ESP Programmer pin", Color.parseColor("#6b40a9"), Color.parseColor("#dc1616")));
+        pinDetails.add(new PinDetails("RXP", "Receiver pin for UART communication", Color.parseColor("#4372a2"), Color.parseColor("#1616dc")));
+        pinDetails.add(new PinDetails("TXP", "Transmitter pin for UART communication", Color.parseColor("#4372a2"), Color.parseColor("#37dc16")));
+        pinDetails.add(new PinDetails("GND", "Ground pin (0 V)", Color.parseColor("#ff4040"), Color.parseColor("#16dcda")));
+        pinDetails.add(new PinDetails("AC1", "Alternative channel input", Color.parseColor("#ffe040"), Color.parseColor("#b01498")));
+        pinDetails.add(new PinDetails("MIC", "External microphone input", Color.parseColor("#7d5840"), Color.parseColor("#890b41")));
```

Comment:
What are these? Did you just copy paste code without checking? These are specific to PSLab

Reply:
Yes, actually I don't know much about the actual hardware circuit nor have any experience in electronics, so i decided to refer same from the PSLab PIN details for now. @CloudyPadmal from where do you suggest I add these specific pin details? Should I take some help from the hardware repo and update like these : 

Part | References
-- | --
C_Feedthrough | C10 C11 C12 C13
Capacitor | C14 C15 C16 C17 C18 C23 C28 C29 C31 C35 C36 C49 C50 C61 C62
CP | C64 C65 C66 C67 C68 C69
C_Feedthrough | C20 C21
Capacitor | C22 C30 C32 C37 C39 C44
Capacitor | C19
Capacitor | C1 C2 C3 C4 C7 C8 C9 C41 C42 C46 C47 C48 C51 C52 C53 C54 C55 C56 C57 C58 C59 C60
Capacitor | C26 C27 C33 C34
Capacitor | C5 C6 C63
Capacitor | C38 C40 C43 C45

Reply:
Oh my goodness. You know what a pin is right?

Reply:
You can see each pin has a small label right next to it in this image right? https://raw.githubusercontent.com/fossasia/neurolab-hardware/master/docs/images/Top.png
Just Google search each name or refer to PSLab app and add that as a small description. Except for AD0~AD3; they are Analog input 0 ~ 3

Reply:
Okay. understood. I was referring from the Neurolab_BOM_sheets :sweat_smile: